src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -392,45 +329,81 @@ EmberAfNodeOperationalCertStatus ConvertToNOCResponseStatus(CHIP_ERROR err)
 bool emberAfOperationalCredentialsClusterRemoveAllFabricsCallback(EndpointId endpoint, app::CommandHandler * commandObj)
 {
     emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Remove all Fabrics");
-    PlatformMgr().ScheduleWork(DoRemoveAllFabrics, 0);
     emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_SUCCESS);
     return true;
 }
 
 bool emberAfOperationalCredentialsClusterAddNOCCallback(EndpointId endpoint, app::CommandHandler * commandObj, ByteSpan NOCArray,
-                                                        ByteSpan IPKValue, NodeId CaseAdminNode, uint16_t AdminVendorId)
+                                                        ByteSpan IPKValue, NodeId adminNodeId, uint16_t adminVendorId)
 {
     EmberAfNodeOperationalCertStatus nocResponse = EMBER_ZCL_NODE_OPERATIONAL_CERT_STATUS_SUCCESS;
 
-    CHIP_ERROR err      = CHIP_NO_ERROR;
-    uint8_t fabricIndex = 0;
+    CHIP_ERROR err          = CHIP_NO_ERROR;
+    FabricIndex fabricIndex = 0;
 
     emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: commissioner has added an Op Cert");
 
+    err = gFabricBeingCommissioned.SetOperationalCertsFromCertArray(NOCArray);
+    VerifyOrExit(err == CHIP_NO_ERROR, nocResponse = ConvertToNOCResponseStatus(err));
+
+    gFabricBeingCommissioned.SetVendorId(adminVendorId);
+
+    err = GetGlobalFabricTable().AddNewFabric(gFabricBeingCommissioned, fabricIndex);
+    VerifyOrExit(err == CHIP_NO_ERROR, nocResponse = ConvertToNOCResponseStatus(err));
+
+    err = GetGlobalFabricTable().Store(fabricIndex);
+    VerifyOrExit(err == CHIP_NO_ERROR, nocResponse = ConvertToNOCResponseStatus(err));
+
+    // We have a new operational identity and should start advertising it.  We
+    // can't just wait until we get network configuration commands, because we
+    // might be on the operational network already, in which case we are
+    // expected to be live with our new identity at this point.
```

Comment:
Agreed that advertisement needs to start now.  But I find this comment misleading.  Operational identity and access to the network are completely orthogonal.  The reason to start advertising now is that once OpCerts are stored, the node has an operational identity.

Reply:
I'll update the comment. It was mostly copy-pasted from `AddNOC`, but I can see how it doesn't apply for `UpdateNOC`.